internal/backend/remote-state/s3/go.mod

```diff
@@ -3,17 +3,17 @@ module github.com/hashicorp/terraform/internal/backend/remote-state/s3
 go 1.23.1
 
 require (
-	github.com/aws/aws-sdk-go-v2 v1.25.3
-	github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.15.3
-	github.com/aws/aws-sdk-go-v2/feature/s3/manager v1.16.9
-	github.com/aws/aws-sdk-go-v2/service/dynamodb v1.30.4
-	github.com/aws/aws-sdk-go-v2/service/s3 v1.51.4
-	github.com/aws/smithy-go v1.20.1
+	github.com/aws/aws-sdk-go-v2 v1.30.5
```

Comment:
If we update our deps, we also need to run `make syncdeps` to align everything with the root module and other remote backends. This will trigger reviews from the other backend maintainers and core. I don't see anything too wild in the shared dependencies (`logr`, `hcl/v2`, etc.) so this should be a fairly easy review.